### ISSUE TITLE
Event Handling

### ISSUE THREAD

**Issue by anywhichway** [krasimir/absurd#113]
First, EXCELLENT WORK. I had taken to writing my own preprocessor and then someone pointed me to AbsurdJS. I subsequently found the article http://code.tutsplus.com/tutorials/absurdjs-or-why-i-wrote-my-own-css-preprocessor--net-36003. All my reasons were the same.

The only thing that puzzles me about AbsurdJS at the moment is what I see as an un-necessarily complex event handling. Why not just use the names of events like any other selector, e.g.

````
{p: { onclick: function() { alert('paragraph clicked') } } }
````

or

````
{clickResponder { onclick: function() { alert('paragraph clicked') } } }
<div class="click-responder"></div>
````

which the prepocessor would morph into:

````
<div class="click-responder" onclick=refToCompiledFunction></div>
````

You can see the start of my work at https://github.com/anywhichway/csxjs for more detail on the above. I will most likely abandon this now in favor of AbsurdJS.

Also, have you looked at JSON5? You might consider a fork and modififications to help clean-up the JSON key quoting problem that makes things a little messy at times.

**Comment by krasimir** [comment]
Thanks for the links @anywhichway Indeed the syntax about event handling is a little bit ccomplex. I'll check JSON5 during the weekend.

P.S.
I'm open to pull requests by the way :)

**Comment by anywhichway** [comment]
I took a look at the Absurd source and I don't see a logical place to retrofit or add in order to support simplified event handling in your current event handling code. The CSXJS code simply walks the DOM elements  based on selectors and assigns functions named the same as valid HTML5 event handlers to the corresponding attributes on the elements. Pretty simple really. Conceptually it would probably belong in your HTML compiler morph function, although perhaps instead of morph("html").add it would be morph("html").activate, e.g.

````
morph("html").activate({body: 
    {p : {onclick: function() { alert("paragraph clicked") } },
    {#1: {onclick: function() { alert("element 1 clicked") }}}).compile(function(err,html) { console.log(html) });

````
would effectively change

````
<body>
<p>p1</p>
<div id="1"></div>
<p>p2</p>
</body>
````

to

````
<body>
<p onclick='function() { alert("paragraph clicked") }'>p1</p>
<div id="1" onclick='function() { alert("element 1 clicked") }'></div>
<p onclick='function() { alert("paragraph clicked") }'>p2</p>
</body>
````

If the above makes sense, I'll poke around a litle more and see if a fork and pull request make sense.

**Comment by krasimir** [comment]
Hi @anywhichway 
thanks for the try. I'll do my best to come with a solution next few days.

**Comment by anywhichway** [comment]
ok, thanks

**Comment by krasimir** [comment]
Hi @anywhichway 

a new version of the library is released - `0.3.4`. I checked the code that we have now and I could say that passing a anonymous function into the HTML template of the component is really difficult to handle. Mainly because the components and the HTML compilation are completely separated processes. Here is what I came up with:
```js
console.clear();
var absurd = Absurd();
absurd.component('MyComp', {
    showText: function(e, str) {
      console.log(str);
    },
    html: {
      input: {
        $click: 'showText:clicked',
        $keyup: 'showText:keyuppped'
      }
    },
    ready: function(dom) {
        this
        // saying where the element
        // will be attached
        .set('parent', dom('body').el) 
        // compiling the css and html
        .populate();
    }
})();
```
The trick is to define the event with `$` in front. Otherwise it is processed as child node.
Here is a jsfiddle to play with:
http://jsbin.com/zudamumuta/1/edit?js,console,output

Let me know what you think ;)

P.S.
The files on the main site are updated too.

**Comment by anywhichway** [comment]
Thanks! Pretty straight forward.
